panoramapublic/src/org/labkey/panoramapublic/pipeline/CopyExperimentFinalTask.java

```diff
@@ -657,9 +661,91 @@ private void assignReader(UserPrincipal reader, Container target, User pipelineJ
         SecurityPolicyManager.savePolicy(newPolicy, pipelineJobUser);
     }
 
-    public static String createPassword()
+    private static class ReviewerAndPassword
+    {
+        private final User _reviewer;
+        private final String _password;
+
+        public ReviewerAndPassword(@NotNull User reviewer, @NotNull String password)
+        {
+            _reviewer = reviewer;
+            _password = password;
+        }
+
+        public User getReviewer()
+        {
+            return _reviewer;
+        }
+
+        public String getPassword()
+        {
+            return _password;
+        }
+    }
+
+    private static String createPassword(User user)
+    {
+        String password;
+        do {
+            password = PasswordGenerator.generate();
+        } while (!PasswordRule.Strong.isValidForLogin(password, user, null));
+
+        return password;
+    }
+
+    private static class PasswordGenerator
     {
-        return RandomStringUtils.randomAlphabetic(8);
+        private static final List<Character> LOWERCASE = "abcdefghijklmnopqrstuvwxyz".chars().mapToObj(c -> (char)c).collect(Collectors.toList());
+        private static final List<Character> UPPERCASE = "ABCDEFGHIJKLMNOPQRSTUVWXYZ".chars().mapToObj(c -> (char)c).collect(Collectors.toList());
+        private static final List<Character> DIGITS = "0123456789".chars().mapToObj(c -> (char)c).collect(Collectors.toList());
+        private static final List<Character> SYMBOLS = "!@#$%^&*+=?".chars().mapToObj(c -> (char)c).collect(Collectors.toList());
+
+        private static final int PASSWORD_LEN = 14;
+
+        public static String generate()
+        {
+            SecureRandom random = new SecureRandom();
+
+            List<Character> passwordChars = new ArrayList<>(PASSWORD_LEN);
+            List<Character> allChars = new ArrayList<>();
+
+            // Initialize the list with all possible characters
+            allChars.addAll(LOWERCASE);
+            allChars.addAll(UPPERCASE);
+            allChars.addAll(DIGITS);
+            allChars.addAll(SYMBOLS);
+
+            // Ensure that there is at least one character from each character category.
+            addChar(LOWERCASE, passwordChars, allChars, random);
+            addChar(UPPERCASE, passwordChars, allChars, random);
+            addChar(DIGITS, passwordChars, allChars, random);
+            addChar(SYMBOLS, passwordChars, allChars, random);
+
+            // Shuffle the list of remaining characters
+            Collections.shuffle(allChars);
+
+            // Add more characters until we are at the desired password length
+            while (passwordChars.size() < PASSWORD_LEN)
+            {
+                addChar(allChars, passwordChars, allChars, random);
+            }
+
+            Collections.shuffle(passwordChars);
+
+            return passwordChars.stream().map(String::valueOf).collect(Collectors.joining());
```

Comment:
It's extremely unlikely, but it's possible this could choose a password that doesn't meet our entropy threshold. You could check the generated value via `PasswordRule.Strong.isValidForLogin()` before returning it.

Reply:
That check is already happening in the `createPassword(User user)` method (line686) of `CopyExperimentFinalTask`.